src/shared/utilities/vsCodeUtils.ts

```diff
@@ -0,0 +1,111 @@
+/*!
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+import * as vscode from 'vscode'
+import * as nls from 'vscode-nls'
+export const localize = nls.loadMessageBundle()
+
+import { ErrorOrString, getLogger, Logger } from '../logger'
+
+// ------- Experimental combined output channel & traditional logger  --------------
+
+type LogLevel = 'debug' | 'info' | 'warn' | 'error'
+
+interface LogParams {
+    args: ErrorOrString[],
+    channel: vscode.OutputChannel,
+    level: LogLevel,
+    logger: Logger,
+    msg: string,
+    nlsKey: string,
+}
+
+// TODO: Leverage 3rd party code: https://raw.githubusercontent.com/Microsoft/vscode-nls/master/src/main.ts
+function defaultTranslation(message: string, ...args: any[]): string {
```

Comment:
It's not clear to me from the title what this function does. Also, to me `default` implies that I can provide something other than the default, which is not the case.  Consider something like `format` or `formatMessage`.

Reply:
✔️